src/ape/managers/project/dependency.py

```diff
@@ -199,3 +202,101 @@ def uri(self) -> AnyUrl:
 
     def extract_manifest(self) -> PackageManifest:
         return self._extract_local_manifest(self.path)
+
+
+
+
+
+class NpmDependency(DependencyAPI):
+    """
+    A dependency frpm NPM. 
+
+    Config example::
+
+        dependencies:
+          - name: Dependency
+            npm: "@gnosis.pm/safe-singleton-factory"
+            version: 1.0.3
+    """
+
+    npm: str
+    """
+    The NPM repo ID e.g. the organization name followed by the repo name,
+    such as ``"@gnosis.pm/safe-singleton-factory"``
+    """
+
+    @cached_property
+    def version_id(self) -> str:
+        if self.version:
+            return self.version
+
+    @property
+    def check_npm(self) -> str:
+        _npm = shutil.which("npm")
+        if not _npm:
+            raise ProjectError(f"Could not locate `npm` executable.")
+        
+        return _npm
+    
+    @property
+    def uri(self) -> AnyUrl:
+        _uri = f"https://github.com/{self.github.strip('/')}"
+        if self.version:
+            version = f"v{self.version}" if not self.version.startswith("v") else self.version
+            _uri = f"{_uri}/releases/tag/{version}"
+        elif self._reference:
+            _uri = f"{_uri}/tree/{self._reference}"
+
+        return HttpUrl(_uri, scheme="https")
+    
+    def _install(*args) -> Tuple[bytes, bytes]:
```

Comment:
```suggestion
    def _communicate(*args) -> Tuple[bytes, bytes]:
```

since is kinda generic in what it does